mautic/mautic
# Unique id fields

Adds a unique field option for all lead fields. Allows different fields to be set as unique identifiers. Now mautic can identify leads by things like twitter, or facebook data, instead of just email addresses. 

Also updates the form submit action to use the new feature in the form submit / update lead action. 

How to test:
+ Create a form which updates a lead via the email address as the identifer. 
+ Checkout the PR 
+ Run the entity migration update script
+ Test the same form action and make sure it doesn't break. Make sure 'email' gets set as a unique identifer by editing the fields. 
+ Also test another field like twitter or facebook. A lead with just a twitter handle should be updatable by adding more form fields. If the user puts in their twitter handle and we already have it, the rest of the lead fields should be updated instead of a new lead being created.
